components/net/image_cache.rs

```diff
@@ -404,6 +414,50 @@ struct RasterizationTask {
     result: Option<RasterImage>,
 }
 
+/// Used for storing images that do not have a `WebRenderImageKey` yet.
+#[derive(Debug, MallocSizeOf)]
+enum PendingKey {
+    RasterImage((LoadKey, RasterImage)),
+    Svg((LoadKey, RasterImage, DeviceIntSize)),
+}
+
+/// The state of the `WebRenderImageKey`` cache
+#[derive(Debug, MallocSizeOf)]
+enum KeyCacheState {
+    /// We already requested a batch of keys.
+    PendingBatch,
+    /// We have some keys in the cache.
+    Ready(VecDeque<WebRenderImageKey>),
```

Comment:
I'm thinking that this can be a `Vec` instead, since we always received a batch as a whole(and are not using `push_back`)?